event/listener.php

```diff
@@ -203,6 +203,25 @@ public function show_idea($event)
 			$this->auth->acl_get('f_vote', (int) $this->config['ideas_forum_id']) &&
 			$event['topic_data']['topic_status'] != ITEM_LOCKED);
 
+		$s_voted_up = $s_voted_down = false;
+		if ($idea['idea_votes_up'] || $idea['idea_votes_down'])
+		{
+			$votes = $this->ideas->get_voters($idea['idea_id']);
+
+			foreach ($votes as $vote)
+			{
+				$this->template->assign_block_vars('votes_' . ($vote['vote_value'] ? 'up' : 'down'), array(
+					'USER' => $vote['user'],
+				));
+
+				if ($this->user->data['user_id'] == $vote['user_id'])
+				{
+					$s_voted_up = ((int) $vote['vote_value'] === 1);
+					$s_voted_down = ((int) $vote['vote_value'] === 0);
```

Comment:
Wouldn’t we just break here once we’ve matched on a vote?

Reply:
No because we still need to assign template vars for each user who voted, same goes for other comment why we need to go through this.